### PR TITLE
sql: Remove potential infinite loops from TestAsyncSchemaChanger

### PR DESCRIPTION
**Pull request by vivekmenezes**
[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4168)

### CODE REVIEW

File: sql/schema_changer_test.go [hunk 1]
```diff
@@ -331,6 +331,9 @@ CREATE INDEX foo ON t.test (v)
 
 	// Wait until index is created.
 	for r := retry.Start(retryOpts); r.Next(); {
+		if r.CurrentAttempt() > 30 {
+			t.Fatalf("made %d attempts checking schema change completion", r.CurrentAttempt())
```
Comment (tamird) [inline sql/schema_changer_test.go]: why don't you set `MaxRetries` in `retryOpts`?

Comment (tamird) [inline sql/schema_changer_test.go]: or use `util.SucceedsWithin`?

**Comment by vivekmenezes** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/schema_changer_test.go, line 335 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4168#-K9nDnpSbo3ftk_uhO1v-r1-335:-K9nKfpsBicAJliZ2xvm:-1059953174) ([raw file](https://github.com/cockroachdb/cockroach/blob/bfb48676fca872a1ff81fe6d0f0e03f2b2b91637/sql/schema_changer_test.go#L335)):*
changed to use util.SucceedsWithin(). Thanks!

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4168)*

**Comment by tamird** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/schema_changer_test.go, line 335 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4168#-K9nDnpSbo3ftk_uhO1v-r1-335:-K9nLiNmgYVstDkRvaqV:1128097230) ([raw file](https://github.com/cockroachdb/cockroach/blob/bfb48676fca872a1ff81fe6d0f0e03f2b2b91637/sql/schema_changer_test.go#L335)):*
It doesn't make sense to use SucceedsWithin as well as retry.Start. You should use on or the other.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4168)*

**Comment by petermattis** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[sql/schema_changer_test.go, line 334 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4168#-K9nMmHJVrRy8X_MXHfJ:-K9nMmHK_3JLRBAd7zRk:1639870069) ([raw file](https://github.com/cockroachdb/cockroach/blob/44e4ff39f7b4a0d159647234f316617d67b99e96/sql/schema_changer_test.go#L334)):*
How does this fix the test? Low timeout limits like this cause flakiness on CircleCI. I prefer to see this test retry until it succeeds or times out.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4168)*

**Comment by vivekmenezes** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[sql/schema_changer_test.go, line 334 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4168#-K9nMmHJVrRy8X_MXHfJ:-K9nNG34cRtawEH7rRyn:-793459538) ([raw file](https://github.com/cockroachdb/cockroach/blob/44e4ff39f7b4a0d159647234f316617d67b99e96/sql/schema_changer_test.go#L334)):*
I was investigating https://github.com/cockroachdb/cockroach/issues/4107
and after reviewing the logs was unable to understand why the test failed.
Neither did the test fail nor timeout. So began suspecting it timed out.

See the link to the artifacts. 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4168)*

**Comment by petermattis** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[sql/schema_changer_test.go, line 334 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4168#-K9nMmHJVrRy8X_MXHfJ:-K9nP7FizkaY5cI2cqmq:1180146618) ([raw file](https://github.com/cockroachdb/cockroach/blob/44e4ff39f7b4a0d159647234f316617d67b99e96/sql/schema_changer_test.go#L334)):*
Sure, but my question stands: how does this fix the test? You're just setting a lower timeout, not actually fixing anything that would have caused the test to not make progress.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4168)*

**Comment by tamird** [comment]
#4166 <https://github.com/cockroachdb/cockroach/issues/4166> fixed #4107
<https://github.com/cockroachdb/cockroach/issues/4107>; the specific test
mentioned appears to be a red herring.

**Comment by vivekmenezes** [comment]
We don't need this. Withdrawing this PR.

**Pull request closed without merging**